src/main/java/com/devcycle/sdk/server/local/bucketing/LocalBucketing.java

```diff
@@ -139,61 +142,63 @@ public void setPlatformData(String platformData) {
         fn.accept(platformDataAddress);
     }
 
-    public BucketedUserConfig generateBucketedConfig(String token, User user) throws JsonProcessingException {
+    public BucketedUserConfig generateBucketedConfig(String sdkKey, User user) throws JsonProcessingException {
         String userString = OBJECT_MAPPER.writeValueAsString(user);
 
-        int tokenAddress = newWasmString(token);
+        int sdkKeyAddress = getSDKKeyAddress(sdkKey);
         int userAddress = newWasmString(userString);
 
         Func generateBucketedConfigForUserPtr = linker.get(store, "", "generateBucketedConfigForUser").get().func();
         WasmFunctions.Function2<Integer, Integer, Integer> generateBucketedConfigForUser = WasmFunctions.func(
                 store, generateBucketedConfigForUserPtr, I32, I32, I32);
 
-        int resultAddress = generateBucketedConfigForUser.call(tokenAddress, userAddress);
+        int resultAddress = generateBucketedConfigForUser.call(sdkKeyAddress, userAddress);
         String bucketedConfigString = readWasmString(resultAddress);
 
         ObjectMapper objectMapper = new ObjectMapper();
         BucketedUserConfig config = objectMapper.readValue(bucketedConfigString, BucketedUserConfig.class);
         return config;
     }
 
-    public void initEventQueue(String token, String options) {
-        int tokenAddress = newWasmString(token);
+    public void initEventQueue(String sdkKey, String options) {
+        int sdkKeyAddress = getSDKKeyAddress(sdkKey);
         int optionsAddress = newWasmString(options);
 
         Func initEventQueuePtr = linker.get(store, "", "initEventQueue").get().func();
         WasmFunctions.Consumer2<Integer, Integer> fn = WasmFunctions.consumer(store, initEventQueuePtr, I32, I32);
-        fn.accept(tokenAddress, optionsAddress);
+        fn.accept(sdkKeyAddress, optionsAddress);
     }
 
-    public void queueEvent(String token, String user, String event) {
-        int tokenAddress = newWasmString(token);
-        int userAddress = newWasmString(user);
+    public void queueEvent(String sdkKey, String user, String event) {
+        int sdkKeyAddress = newWasmString(sdkKey);
+        int userAddress = getPinnedParameter(user);
         int eventAddress = newWasmString(event);
 
         Func queueEventPtr = linker.get(store, "", "queueEvent").get().func();
         WasmFunctions.Consumer3<Integer, Integer, Integer> fn = WasmFunctions.consumer(store, queueEventPtr, I32, I32, I32);
-        fn.accept(tokenAddress, userAddress, eventAddress);
+        fn.accept(sdkKeyAddress, userAddress, eventAddress);
+        unpinParameter(userAddress);
```

Comment:
what happens in case of errors thrown above?

Reply:
all errors are surfaced in the `DVCLocalClient`

Reply:
yeah but the values wont be unpinned until the next time one of the two methods that triggers unpin is called.

the approach I went with for c# was just to unpin at the beginning of every method so that you'd only have at most one method call's worth of pinned resources left behind by some error

Reply:
awesome, will do the same here :) thanks